### PR TITLE
Update varnish 6 vcl config

### LINKED ISSUE #31445: [Issue] Update varnish 6 vcl config
This issue is automatically created based on existing pull request: magento/magento2#31330: Update varnish 6 vcl config

---------

### Description (*)

Magento 2.4-develop

Update varnish logic bypass for customer page and health check request

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. Configure Varnish 6;
2. Go to Storefront;

### Expected Result: (*)

The customer page should not be cached.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Update varnish logic bypass for customer page and health check request

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#31445: Update varnish 6 vcl config

### CODE REVIEW

File: app/code/Magento/PageCache/etc/varnish6.vcl [hunk 1]
```diff
@@ -62,13 +55,13 @@ sub vcl_recv {
         return (pass);
     }
 
-    # Bypass shopping cart and checkout
-    if (req.url ~ "/checkout") {
+    # Bypass customer, shopping cart, checkout and search requests
+    if (req.url ~ "/customer" || req.url ~ "/checkout" || req.url ~ "/catalogsearch") {
```
Comment (kandy) [inline app/code/Magento/PageCache/etc/varnish6.vcl]: catalogsearch now is cacheable, why you skip it?

File: app/code/Magento/PageCache/etc/varnish6.vcl [hunk 2]
```diff
@@ -62,13 +55,13 @@ sub vcl_recv {
...
         return (pass);
     }
 
     # Bypass health check requests
-    if (req.url ~ "/pub/health_check.php") {
+    if (req.url ~ "^/(pub/)?(health_check.php|health.php)$") {
```
Comment (kandy) [inline app/code/Magento/PageCache/etc/varnish6.vcl]: Magento codebase does not have `health.php` file

File: app/code/Magento/PageCache/etc/varnish6.vcl [hunk 3]
```diff
@@ -9,13 +9,6 @@ backend default {
     .host = "/* {{ host }} */";
     .port = "/* {{ port }} */";
     .first_byte_timeout = 600s;
-    .probe = {
```
Comment (kandy) [inline app/code/Magento/PageCache/etc/varnish6.vcl]: This checks need for grace functionality to work, see https://varnish-cache.org/docs/trunk/users-guide/vcl-grace.html

**Comment by mrtuvn** [comment]
@kandy updated as your review! Thanks for your comment

**Comment by mrtuvn** [comment]
@magento run all tests

**Review by kandy — APPROVED** [review]

**Comment by mrtuvn** [comment]
@magento run Semantic Version Checker, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-8554](https://jira.corp.magento.com/browse/ENGCOM-8554) has been created to process this Pull Request
:eight_spoked_asterisk: @orlangur, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@magento create issue

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Configure Varnish 6;
2. Go to Storefront->My Account page 

**Before:** :heavy_multiplication_x: 


**After:** :heavy_check_mark: The Customer page is not cached.

**Pull request merged**